service/tcp.go

```diff
@@ -158,18 +160,27 @@ type streamHandler struct {
 }
 
 // NewStreamHandler creates a StreamHandler
-func NewStreamHandler(authenticate StreamAuthenticateFunc, timeout time.Duration) StreamHandler {
+func NewStreamHandler(authenticate StreamAuthenticateFunc, timeout time.Duration, dialer transport.StreamDialer) StreamHandler {
 	return &streamHandler{
 		readTimeout:  timeout,
 		authenticate: authenticate,
-		dialer:       defaultDialer,
+		dialer:       dialer,
 	}
 }
 
-var defaultDialer = makeValidatingTCPStreamDialer(onet.RequirePublicIP)
-
-func makeValidatingTCPStreamDialer(targetIPValidator onet.TargetIPValidator) transport.StreamDialer {
+func MakeValidatingTCPStreamDialer(targetIPValidator onet.TargetIPValidator, fwmark uint) transport.StreamDialer {
```

Comment:
Unfortunately `SO_MARK` is Linux-specific, not POSIX: https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_10_16

Our server can run on any platform, but your change is making it Linux-only. We do run it on macOS, for instance, for development.

We will need to change the code to make the SO_MARK code only triggers on Linux, so that we can still build it at least for Windows and macOS.

But we can take care of that. For now, please move all the Linux-specific code into the main package, so that we can keep using the shared code across platforms.